src/odemis/util/spectrum.py

```diff
@@ -176,6 +179,78 @@ def get_time_range(data):
         return list(range(min_t, max_t + 1)), "px"
 
 
+def get_angle_range(data):
+    """ Returns the list of theta values, if exists, in degrees.
+
+    :param data: (model.DataArray of shape A...): the DataArray with metadata
+        MD_THETA_LIST
+
+    :return: (list of numbers or None), unit string:
+    """
+
+    try:
+        return get_angle_per_pixel(data), "°"  # degree symbol
+    except (ValueError, KeyError):
+        dims = data.metadata.get(model.MD_DIMS, "CTZYX"[-data.ndim:])
+
+        if len(dims) == 3 and dims == "YXC" and data.shape[2] in (3, 4):  # RGB?
+            # This is a hack that works the same way as in get_wavelength_per_pixel
+            thetai = 0
+        else:
+            try:
+                thetai = dims.index("A")  # get index of dimension A
+            except ValueError:
+                raise ValueError("Dimension 'A' not in dimensions, so skip computing time list.")
+
+        # Shows pixels values (ex: 0 -> +50 px), if theta list is not present
+        max_t = data.shape[thetai] // 2  # array of shape AC
+        min_t = (max_t - data.shape[thetai]) + 1
+        return list(range(min_t, max_t + 1)), "px"
+
+
+def get_angle_per_pixel(da):
+    """
+    Computes the angle list for each pixel along the A dimension. The metadata has stored the theta list in radians,
+    but it's converted into degrees for displaying purposes.
+
+    :param da: (model.DataArray of shape A): the DataArray with metadata MD_THETA_LIST
+    :return: (list of float of length A): the theta values (in degrees) for each pixel in A
+    Note that the theta list consists of NaN values, that later are to be removed for display
+    and the radians will be converted to degrees.
+
+    :raises:
+        AttributeError: if no metadata is present
+        KeyError: if MD_THETA_LIST is not available
+        ValueError: if the metadata doesn't provide enough information
+    """
+
+    if not hasattr(da, 'metadata'):
+        raise AttributeError("No metadata found in data array")
+
+    if model.MD_THETA_LIST in da.metadata:
+        thetal = da.metadata[model.MD_THETA_LIST]
+
+        # check available dimension of data
+        dims = da.metadata.get(model.MD_DIMS, "CAZYX"[-da.ndim:])
+
+        if len(dims) == 3 and dims == "YXC" and da.shape[2] in (3, 4):  # RGB data
+            thetai = 0
+        else:
+            try:
+                thetai = dims.index("A")  # get index of dimension A
+            except ValueError:
+                raise ValueError("Dimension 'A' not in dimensions, so skip computing theta list.")
+
+        if len(thetal) != da.shape[thetai]:
+            raise ValueError("Length of theta list does not match length of theta data.")
+
+        # Converts the theta values from radians to degrees wherever there is not a NaN value
+        theta_list = [math.degrees(theta) if str(theta) != 'nan' else numpy.nan for theta in thetal]
```

Comment:
No, this function should return the THETA_LIST in rad. In the Viewport (GUI), you can convert from rad to degrees. You can probably just move this line to just after `get_angle_range()`.
BTW, math.degrees(NaN) returns NaN, so you can keep it simple:
theta_list = [math.degrees(theta) for theta in theta_list]